salt/client/__init__.py

```diff
@@ -258,7 +258,9 @@ def _check_pub_data(self, pub_data):
             )
 
         # Failed to connect to the master and send the pub
-        if 'jid' not in pub_data:
+        if 'error' in pub_data:
+            print(pub_data['error'])
```

Comment:
Because the client is a CLI tool and when it was designed, the decision was made to just print directly from here. It's mostly a legacy decision and we could revisit it but I'm not sure why we would need to.